Split about and support information in features

jeddy3 opened the pull request:
Hello. Thank you for putting together this site. It's such a useful destination to learn about the new stuff in CSS!

When I was looking through it I found the links and badges below the description of each feature a bit disjointed. I think because:

- the combination of internal and external links
- the combination of badges and text links
- the preset badge is related to the polyfill listed at the bottom of the feature 

This pull request loses the badges and splits each feature into two sections:

- About
  - Stage
  - Title
  - Description
  - Example 
  - Spec and Doc links (if the above info isn't enough and a person wants to learn more)
- Support
  - Browsers
  - Polyfills and plugins (and whether the PostCSS plugin is included in Preset Env).

It also removes the redundancy of the staging badge in favour of adding an anchor link to the stage image.

Goes from this:

<img width="705" alt="Screenshot 2022-01-31 at 17 42 40" src="https://user-images.githubusercontent.com/808227/152008838-69a46fc5-236d-456c-9d37-85eedb587b33.png">

To this:

<img width="793" alt="Screenshot 2022-02-01 at 16 01 12" src="https://user-images.githubusercontent.com/808227/152008894-c0048464-92c5-484a-b6fe-a438f8207a21.png">

It's not quite as compact, but the grouping of related information feels better. What do you think? 

Behind the scenes, this pull request moves the code related to the present env into the polyfills components.

Antonio-Laguna commented:
@jeddy3 thanks a lot for taking the time to submit this!

That said, we've been reviewing this internally and we would like to keep some things as they currently are.

We kind of like the badges as they're immediately recognizable and they have this "README" vibe. It might change in the future. The Feature badge that links to the Feature down on the page is used within the READMEs of those plugins so having them visible also helps us ensure that they'll work. See for example [PostCSS Focus Visible](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-focus-visible#postcss-focus-visible-).

This also introduces some accessibility concerns given that there are too many links with the same name but different purposes (this could be fixed).

I think (if you're open to it) we could repurpose this a bit.

- [ ] Update "Specification" to use this badge ![](https://img.shields.io/badge/Spec-005A9C?logo=w3c&style=flat-square) `https://img.shields.io/badge/Spec-005A9C?logo=w3c&style=flat-square`. Let's make sure this has a similar treatment to what the postcss preset env badge has.
- [ ] Restore the badges except for Bundled with PostCSS Preset Env. I think this makes sense to do as you suggest. However, see if you can fix the stars count that now lives next to Preset Env instead of next to the plugin.

jeddy3 commented:
Thanks for the feedback. I didn't know about the W3C spec badge; that's a much better solution!

I've:
- readded the badges 
- kept the bundled-with-preset-env link in the polyfills section
- moved the stars next to the plugin

I've used a negative margin and removed the fixed `inline-size` for the last one so that the space it takes up is dictated by the length of the number.

Looks like this now:

<img width="676" alt="Screenshot 2022-02-03 at 17 08 31" src="https://user-images.githubusercontent.com/808227/152395111-6d6a0008-55f2-42f5-ae78-f47fb6adc3b2.png">

Keeps the README feel, while grouping the polyfill stuff together.

romainmenke commented on src/styles/style.css:
```diff
@@ -311,16 +291,19 @@ img {
 		& img {
 			height: 20px;
-			transform: translateX(-58px);
+			margin-inline-start: -56px;
```
```suggestion
			margin-inline-start: -56px;
			min-width: 57px;
```

To make image lazy loading work.
Stars did not appear for me in latest Chrome.

Antonio-Laguna commented:
@jeddy3 thanks a lot! This is great! 

I'm having issues seeing the stargazes at some point. At least on Chrome, they don't work but if I mess with the margin they just show up.
![Feb-04-2022 11-02-14](https://user-images.githubusercontent.com/946645/152510049-e5ac3b3e-eb94-4238-ac41-e2c14a47e0d6.gif)

@romainmenke has been debugging t and  It looks like this has to do with lazy loading and the margin making it virtually invisible so the browser thinks it's not there. Some min-width on the image should make it show up.

Antonio-Laguna commented:
Thanks for all of the effort @jeddy3 !

Merging this now!
